mne/io/_read_raw.py

```diff
@@ -0,0 +1,80 @@
+"""Generic wrapper function read_raw for specific read_raw_xxx readers."""
+
+# Authors: Clemens Brunner <clemens.brunner@gmail.com>
+#
+# License: BSD (3-clause)
+
+
+from pathlib import Path
+from functools import partial
+
+from . import (read_raw_edf, read_raw_bdf, read_raw_gdf, read_raw_brainvision,
+               read_raw_fif, read_raw_eeglab, read_raw_cnt, read_raw_egi,
+               read_raw_eximia, read_raw_nirx, read_raw_fieldtrip,
+               read_raw_artemis123, read_raw_nicolet, read_raw_kit,
+               read_raw_ctf)
+
+
+def _read_unsupported(fname, **kwargs):
+    ext = "".join(Path(fname).suffixes)
+    msg = f"Unsupported file type ({ext})."
+    suggest = kwargs.get("suggest")
+    if suggest is not None:
+        msg += f" Try reading a {suggest} file instead."
+    msg += " Consider using a dedicated reader function for more options."
+    raise ValueError(msg)
+
+
+# supported read file formats
+supported = {".edf": read_raw_edf,
+             ".bdf": read_raw_bdf,
+             ".gdf": read_raw_gdf,
+             ".vhdr": read_raw_brainvision,
+             ".fif": read_raw_fif,
+             ".fif.gz": read_raw_fif,
+             ".set": read_raw_eeglab,
+             ".cnt": read_raw_cnt,
+             ".mff": read_raw_egi,
+             ".nxe": read_raw_eximia,
+             ".hdr": read_raw_nirx,
+             ".mat": read_raw_fieldtrip,
+             ".bin": read_raw_artemis123,
+             ".data": read_raw_nicolet,
+             ".sqd": read_raw_kit,
+             ".ds": read_raw_ctf}
+
+# known but unsupported file formats
+suggested = {".vmrk": partial(_read_unsupported, suggest=".vhdr"),
+             ".eeg": partial(_read_unsupported, suggest=".vhdr")}
+
+# all known file formats
+readers = {**supported, **suggested}
+
+
+def read_raw(fname, *, preload=False, verbose=None, **kwargs):
+    """Read raw file.
+
+    Parameters
+    ----------
+    fname : str
+        File name to load.
```

Comment:
add doc for preload and verbose

Reply:
How exactly does this Sphinx `%(preload)s` business work? It doesn't seem to work in this function so I'm guessing I have to add it to some configuration.

Reply:
You need to add the `@verbose` or `@fill_doc` decorator to the function, these do `%` formatting using our `docdict`. You can just use the latter because each reader will have its own `@verbose` and there is no reason to apply it twice

Reply:
Thanks @larsoner!